src/c-writer.cc

```diff
@@ -1638,7 +1695,7 @@ void CWriter::WriteExports(WriteExportsKind kind) {
         const Table* table = module_->GetTable(export_->var);
         mangled_name = ExportName(MangleName(export_->name));
         internal_name = table->name;
-        WriteTablePtr(mangled_name);
+        WriteTablePtr(mangled_name, GetReferenceTypeName(table->elem_type));
```

Comment:
Can we have WriteTablePtr take the table itself as arg2 and have that function to the `GetReferenceTypeName` call?

Reply:
Sure, done.